aeloggingstackdriver.go

```diff
@@ -118,32 +118,47 @@ func getLogCtxVal(aeInfo AppengineInfo, hreq *http.Request, logger *logging.Logg
 }
 
 // for use in flex services with long-running tasks that don't handle http requests
-func WrapBackgroundContextWithStackdriverLogger(c context.Context, logName string) (context.Context, func()) {
+func WrapBackgroundContextWithStackdriverLogger(c context.Context, logName string) context.Context {
 	if IsDevAppServer {
-		return c, func() {}
+		return c
 	}
+	return wrapBackgroundContextWithStackdriverLogger(c, logName, GetOrCreateLoggingClient())
+}
 
+func wrapBackgroundContextWithStackdriverLogger(c context.Context, logName string, lc *logging.Client) context.Context {
 	aeInfo := NewAppengineInfoFromContext(c)
 
 	project := aeInfo.NativeProjectID()
 	if project == "" {
 		panic("aelog: no GCP project set in environment")
 	}
 	parent := "projects/" + project
-	lc := GetOrCreateLoggingClient()
 	if logName == "" {
 		logName = ChildLogName
 	}
 	req, err := http.NewRequest(http.MethodGet, "pendo.io/background", bytes.NewReader([]byte{}))
 	if err != nil {
 		panic(err)
 	}
-	logger := getLogger(aeInfo, lc, logName)
-	logCtxVal := getLogCtxVal(aeInfo, req, logger, parent+"/traces/"+fmt.Sprintf("%d", rand.Int63()))
 
-	ctx := context.WithValue(c, loggingCtxKey, logCtxVal)
+	return context.WithValue(c, loggingCtxKey, getLogCtxVal(aeInfo, req, getLogger(aeInfo, lc, logName), parent+"/traces/"+fmt.Sprintf("%d", rand.Int63())))
+}
+
+func WrapBackgroundContextWithStackdriverLoggerAndClosure(c context.Context, logName string) (context.Context, func()) {
```

Comment:
Let's call this (breathes deeply) `WrapBackgroundContextWithStackdriverLoggerWithCloseFunc` because you're not really returning a closure. (I mean you are, but that's not really descriptive of what it does.)